File: 01_Language/01_Functions/python/array_shift.py
# coding: utf-8

def array_shift(arr):
    try:
        return arr.pop(0)
    except IndexError:
        return None


if __name__ == '__main__':
    stack = ["orange", "banana", "apple", "raspberry"]
    print(array_shift(stack))

    stack = {"k1": "orange", "k2": "banana", 0: "apple", 1: "raspberry"}
    print(array_shift(stack))

File: 01_Language/01_Functions/php/array_shift.php
<?php

$stack = array("orange", "banana", "apple", "raspberry");
$fruit = array_shift($stack);
print_r($stack);

$stack = array("k1" => "orange", "k2" => "banana", "apple", "raspberry");
$fruit = array_shift($stack);
print_r($stack);
